### PR TITLE
commit to work with node 18

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
Since the `pages-e2e`s require node 18 to work I think we need to switch to that for when working on the packages

the packages themselves can still just require 16 (since, I'd imagine that requiring 18 would be considered a breaking change)

I'm not 100% sure if this is the right call 🤔

### CODE REVIEW

File: .github/workflows/beta-release.yml [hunk 1]
```diff
@@ -16,10 +16,10 @@ jobs:
       - name: Checkout Repo
         uses: actions/checkout@v3
 
-      - name: Setup Node.js 16.x
+      - name: Setup Node.js 18.x
```
Comment (GregBrimble) [inline .github/workflows/beta-release.yml:19]:
```suggestion
      - name: Setup Node.js
```

File: .github/workflows/beta-release.yml [hunk 2]
```diff
@@ -16,10 +16,10 @@ jobs:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 16.x
+          node-version: 18.x
```
Comment (GregBrimble) [inline .github/workflows/beta-release.yml:22]:
```suggestion
          node-version-file: package.json
```

File: .github/workflows/create-pullrequest-prerelease.yml [hunk 3]
```diff
@@ -14,10 +14,10 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Use Node.js 16.13
+      - name: Use Node.js 18.x
```
Comment (GregBrimble) [inline .github/workflows/create-pullrequest-prerelease.yml:17]:
```suggestion
      - name: Use Node.js
```

File: .github/workflows/create-pullrequest-prerelease.yml [hunk 4]
```diff
@@ -14,10 +14,10 @@ jobs:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 16.13
+          node-version: 18.x
```
Comment (GregBrimble) [inline .github/workflows/create-pullrequest-prerelease.yml:20]:
```suggestion
          node-version-file: package.json
```

File: .github/workflows/pull-request.yml [hunk 5]
```diff
@@ -16,7 +16,7 @@ jobs:
       - name: Setup node
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:16]:
```suggestion
      - name: Setup Node.js
```

File: .github/workflows/pull-request.yml [hunk 6]
```diff
@@ -16,7 +16,7 @@ jobs:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 16.13.x
+          node-version: 18.x
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:19]:
```suggestion
          node-version-file: package.json
```

File: .github/workflows/pull-request.yml [hunk 7]
```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Setup node
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:32]:
```suggestion
      - name: Setup Node.js
```

File: .github/workflows/pull-request.yml [hunk 8]
```diff
@@ -32,7 +32,7 @@ jobs:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 16.13.x
+          node-version: 18.x
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:35]:
```suggestion
          node-version-file: package.json
```

File: .github/workflows/pull-request.yml [hunk 9]
```diff
@@ -48,7 +48,7 @@ jobs:
       - name: Setup node
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:48]:
```suggestion
      - name: Setup Node.js
```

File: .github/workflows/pull-request.yml [hunk 10]
```diff
@@ -48,7 +48,7 @@ jobs:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 16.13.x
+          node-version: 18.x
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:51]:
```suggestion
          node-version-file: package.json
```

File: .github/workflows/pull-request.yml [hunk 11]
```diff
@@ -64,7 +64,7 @@ jobs:
       - name: Setup node
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:64]:
```suggestion
      - name: Setup Node.js
```

File: .github/workflows/pull-request.yml [hunk 12]
```diff
@@ -64,7 +64,7 @@ jobs:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 16.13.x
+          node-version: 18.x
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:67]:
```suggestion
          node-version-file: package.json
```

File: .github/workflows/pull-request.yml [hunk 13]
```diff
@@ -80,7 +80,7 @@ jobs:
       - name: Setup node
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:80]:
```suggestion
      - name: Setup Node.js
```

File: .github/workflows/pull-request.yml [hunk 14]
```diff
@@ -80,7 +80,7 @@ jobs:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 16.13.x
+          node-version: 18.x
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:83]:
```suggestion
          node-version-file: package.json
```

Comment (GregBrimble) [reply .github/workflows/pull-request.yml:83]: And fix in the e2e section too please!

File: .github/workflows/pull-request.yml (hunk 13 above)
Comment (GregBrimble) [reply .github/workflows/pull-request.yml:80]: And fix in the e2e section too please!

File: .github/workflows/release.yml [hunk 15]
```diff
@@ -16,10 +16,10 @@ jobs:
       - name: Checkout Repo
         uses: actions/checkout@v3
 
-      - name: Setup Node.js 16.x
+      - name: Setup Node.js 18.x
```
Comment (GregBrimble) [inline .github/workflows/release.yml:19]:
```suggestion
      - name: Setup Node.js
```

File: .github/workflows/release.yml [hunk 16]
```diff
@@ -16,10 +16,10 @@ jobs:
...
         uses: actions/setup-node@v3
         with:
-          node-version: 16.x
+          node-version: 18.x
```
Comment (GregBrimble) [inline .github/workflows/release.yml:22]:
```suggestion
          node-version-file: package.json
```

File: .npmrc [hunk 17]
```diff
@@ -1 +1,3 @@
 //registry.npmjs.org/:_authToken=${NPM_TOKEN}
+
+engine-strict=true
```
Comment (GregBrimble) [inline .npmrc:3]: Why is this necessary?

File: package.json [hunk 18]
```diff
@@ -19,10 +19,13 @@
 		"publish": "turbo build & changeset publish",
 		"changeset": "npx changeset"
 	},
+	"engines": {
+		"node": ">=18"
```
Comment (GregBrimble) [inline package.json:23]: Should this be tighter? You want people to run exactly Node.js 18, no? Or are you happy with people running 20?

File: pages-e2e/package.json [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+	},
+	"dependencies": {
+		"@cfpreview/pages-e2e-test-runner-cli": "^0.0.20",
+		"wrangler": "beta"
+	},
+	"volta": {
+		"node": "18.16.1"
+	},
+	"engines": {
+		"node": "18.16.1"
```
Comment (GregBrimble) [inline pages-e2e/package.json:15]:
```suggestion
```

No longer necessary.

File: package.json [hunk 20]
```diff
@@ -19,10 +19,13 @@
...
+	},
```
Comment (GregBrimble) [inline package.json:24]:
```suggestion
	},
	"volta": {
	  "node": "18.16.1"
	},
```

File: .npmrc (hunk 17 above)
Comment (dario-piotrowicz) [reply .npmrc:3]: it's not necessary, I can remove it if you don't like it, I thought it was a nice touch to force people to have the correct node version instead of just warning about it 🙂

File: pages-e2e/package.json (hunk 19 above)
Comment (dario-piotrowicz) [reply pages-e2e/package.json:15]: yeah, sorry I missed it 😓

File: package.json (hunk 20 above)
Comment (dario-piotrowicz) [reply package.json:24]: why? 😮

File: package.json (hunk 18 above)
Comment (dario-piotrowicz) [reply package.json:23]: mh.... I don't know.... do we? do you see any problem if people were to use 20? (considering that we use 18 in the CI checks)

File: package.json
Comment (james-elicx) [inline package.json:1]: The `test:e2e` script in the root `package.json` needs to be changed to run the `test` script in the `@cloudflare/pages-e2e` workspace.

It is currently erroring in the workflow, even though the workflow is showing success.

image

File: package.json (hunk 20 above)
Comment (GregBrimble) [reply package.json:24]: For folks who use Volta (like me! 😄)

File: package.json
Comment (dario-piotrowicz) [reply package.json:1]: wow.... I can't believe that passes 😕

File: package.json (hunk 18 above)
Comment (GregBrimble) [reply package.json:23]: 20 (or future versions) could have breaking changes, so I think restricting to `^18` is probably correct, but up to you.

File: .github/workflows/create-pullrequest-prerelease.yml [hunk 21]
```diff
@@ -14,10 +14,10 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Use Node.js 16.13
+      - name: Use Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: 16.13
+          node-version-file: package.json
           cache: 'npm' # cache ~/.npm in case 'npm ci' needs to run
```
Comment (GregBrimble) [inline .github/workflows/create-pullrequest-prerelease.yml:21]: None of the other steps include this. Should we include it?

Comment (dario-piotrowicz) [reply .github/workflows/create-pullrequest-prerelease.yml:21]: ah... I'd remove it and we can put these back in a separate PR 🙂

File: .github/workflows/pull-request.yml [hunk 22]
```diff
@@ -77,10 +77,10 @@ jobs:
...
         uses: actions/checkout@v3
-      - name: Setup node
+      - name: Setup Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: 16.13.x
+          node-version-file: package.json
 
       - name: Install
         run: npm ci
```
Comment (GregBrimble) [inline .github/workflows/pull-request.yml:86]: Can you fix L96–99 too please?

**Review by GregBrimble — APPROVED** [review]

File: .github/workflows/pull-request.yml (hunk 22 above)
Comment (dario-piotrowicz) [reply .github/workflows/pull-request.yml:86]: on it! 🙂

Comment (dario-piotrowicz) [reply .github/workflows/pull-request.yml:86]: done 🙂

File: .github/workflows/create-pullrequest-prerelease.yml (hunk 21 above)
Comment (dario-piotrowicz) [reply .github/workflows/create-pullrequest-prerelease.yml]: removed 🙂

File: package.json (hunk 18 above)
Comment (dario-piotrowicz) [reply package.json]: I don't have a huge preference, I wouldn't want to bother people to force them to install `18` if they for example have `20`, but let's go for `18` and we can add `20` or whatever else later on if we want 🙂

File: package.json [hunk 23]
```diff
@@ -4,7 +4,7 @@
 	"workspaces": [
 		"internal-packages/*",
 		"packages/*",
-		"pages-e2e/features/*"
+		"pages-e2e/*"
```
[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply package.json:7]: why? 😮 

I wanted to add both `pages-e2e` but also keep apps inside `pages-e2e/features` as Greg set (so that on `npm i` we get everything I guess?)

without the `/*` we would only get the `pages-e2e` app, wouldn't we?

[the remark this answers is not in the document]
Comment (james-elicx) [reply package.json:7]: it doesn't catch the workspace in `pages-e2e` at the moment since it's only looking at subdirectories, so turbo isnt running the command in any workspaces for `test:e2e`. you could put both `pages-e2e` and `pages-e2e/features/*` (from before) if you want both i guess

[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply package.json:7]: ok it should be `/**` 😅 

but with that I get:
Screenshot 2023-07-03 at 21 24 32

so let's go with 
```
"pages-e2e",
"pages-e2e/features/*"
```

Thank a lot for noticing it! (stupid me for not testing it locally! 🤦)

[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply package.json]: I have to look into this.... it looks like whatever dependencies I add to the fixtures don't get saved to the lock file nor are they present in the node_modules... 😕

[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply package.json]: I think npm doesn't recognize workspaces inside a workspace 😓

[the remark this answers is not in the document]
Comment (james-elicx) [reply package.json]: > I have to look into this.... it looks like whatever dependencies I add to the fixtures don't get saved to the lock file nor are they present in the node_modules... 😕

i think there may be a typo in one of the new additions that could be the cause of that

[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply package.json]: ok that works, but it has the effect of not generating a lock file for the fixture

@GregBrimble is this acceptable for the pages-e2es? or do the app needs to have a lock file? (I think it does but maybe I'm wrong?)

**Comment by dario-piotrowicz** [comment]
I'm closing this PR as we are probably not going forward with this (and instead stick with node 16 for the time being)
